### PR TITLE
initialize mesh when asset is loaded

### PR DESCRIPTION
**Pull request by SantyWang**
Re: https://github.com/cocos-creator/3d-tasks/issues/8950

Changelog:
 *

### CODE REVIEW

File: cocos/3d/assets/mesh.ts [hunk 1]
```diff
@@ -1012,10 +1016,6 @@ export class Mesh extends Asset {
             data: globalEmptyMeshBuffer,
         });
     }
-
-    public validate () {
-        return this.renderingSubMeshes.length > 0 && this.data.byteLength > 0;
-    }
```
Comment (pandamicro) [inline cocos/3d/assets/mesh.ts:1018]: Why remove `validate` function ? And which problem this PR wants to solve

**Review by shrinktofit — APPROVED** [review]

File: cocos/3d/assets/mesh.ts (hunk 1 above)
Comment (SantyWang) [reply cocos/3d/assets/mesh.ts:1018]: Can not validate mesh correctly, so move initialize function to `onloaded` and use `try-catch` to validate it

Comment (shrinktofit) [reply cocos/3d/assets/mesh.ts:1018]: This validation is not accurate and groundless, it broke empty meshes.

**Review by pandamicro — APPROVED** [review]

File: cocos/3d/assets/mesh.ts (hunk 1 above)
Comment (dumganhar) [reply cocos/3d/assets/mesh.ts:1018]: If using JS try-catch to validate, how to do it in C++?

**Pull request merged**